signature/sign.go

```diff
@@ -113,16 +120,28 @@ func Sign(_ context.Context, key jwk.Key, sf SignedFielder, opts ...Option) (*pi
 		return nil, err
 	}
 
-	if options.logger != nil {
+	switch key := key.(type) {
+	case jwk.Key:
 		pk, err := key.PublicKey()
 		if err != nil {
 			return nil, fmt.Errorf("unable to generate public key: %w", err)
 		}
+
 		fingerprint, err := pk.Thumbprint(crypto.SHA256)
 		if err != nil {
 			return nil, fmt.Errorf("calculating key thumbprint: %w", err)
 		}
+
 		debug(options.logger, "Public Key Thumbprint (sha256): %s", hex.EncodeToString(fingerprint))
+	case crypto.Signer:
+		data, err := x509.MarshalPKIXPublicKey(key.Public())
+		if err != nil {
+			return nil, fmt.Errorf("failed to marshal public key: %w", err)
+		}
+
+		debug(options.logger, "Public Key Thumbprint (sha256): %x", sha256.Sum256(data))
```

Comment:
While we're here, we could replace `hex.EncodeToString` with a `%x` verb in the other debug log (line 135)